src/main/java/com/napier/sem/UserPrompt.java

```diff
@@ -0,0 +1,318 @@
+     * as a numbered list, and asks the user which one they'd like to choose
+     * asks the
+     * @param parentQuery The ID of the type of query being run
+     * @return The string representing the area filter
+     */
+    private String obtainAreaFilterChoice(int parentQuery)
+    {
+        if (parentQuery == 9) return "";
+
+        QueryInfo queryInfo = queryTable.get(parentQuery);
+        String[] areaFilterDescriptions = queryInfo.getAreaFilterDescriptions();
+
+        StringBuilder childQueryPrompt = new StringBuilder(String.format(App.HORIZONTAL_LINE + "\n" +
+                        "Choose the type of \"%s\" query you'd like to run.\n",
+                queryInfo.getQueryDescription()));
+        for (int i = 0; i < queryInfo.getAreaFilterDescriptions().length; i++)
+        {
+            childQueryPrompt.append(String.format("%d) %s %s" +
+                            ((i == queryInfo.getAreaFilterDescriptions().length - 1) ? "" : "\n"),
+                    i + 1,
+                    queryInfo.getQueryDescription(),
+                    areaFilterDescriptions[i]));
+        }
+
+        int childQuery = obtainInputWithPrompt(childQueryPrompt.toString(), areaFilterDescriptions.length);
+
+        return parseQueryInputForAreaFilter(parentQuery, childQuery);
+    }
+
+    /**
+     * Display a given string as a prompt for a user to select from a numbered list, and then take an
+     * integer as input from the user.  If the input is not an integer or is larger than the maximum
+     * passed in as an argument, the prompt is displayed again and the user is given an error message.
+     * Will return -1 if the user enters "q", indicating they want to quit.
+     * @param prompt A string that will be printed in the console explaining what the user should enter
+     * @param maxNum An integer representing the largest number a user should be able to enter
+     * @return An integer representing the user's choice
+     */
+    private int obtainInputWithPrompt(String prompt, int maxNum)
+    {
+        int validInput = 0;
+        while (validInput == 0)
+        {
+            System.out.println(prompt);
+            Integer input = obtainIntWithinRange(maxNum);
+            if (input == null)
+            {
+                System.out.println("Sorry, that's not a valid choice. Pick a number from 1 to " + maxNum);
+            }
+            else
+            {
+                validInput = input;
+            }
+        }
+        return validInput;
+    }
+
+    /**
+     * Scans input for a number from 1 to the maxNum, and returns null if there is no matching input.
+     * Returns -1 if the user enters "q", indicating they want to quit.
+     * @param maxNum An integer representing the largest number which should be accepted as valid input
+     * @return An integer from 1 to maxNum, -1 if the user enters "q", or null if no valid input is found
+     */
+    private Integer obtainIntWithinRange(int maxNum)
```

Comment:
the difference between using `int` and `Integer` is actually important here, as if you use `Integer` then you can return `null`, but not if you use `int`